common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -150,13 +158,19 @@ public static boolean isEnabled() {
      * Sets the resource leak detection level.
      */
     public static void setLevel(Level level) {
+        if (stableLevel != null) {
+            return;
```

Comment:
I think we should better throw in this case.

Reply:
a stackless Unsupported Ex? 
I'm saying that because maybe users don't want to have some special paths to handle when the level is "immutable"

Reply:
not sure what is better... At least it wouldn't be surprising in a sense that while they think they use paranoid (for example) it not actually does.

Reply:
Me neither, be silent doesn't seem right, I agree...a warn?

Reply:
I have noticed https://github.com/eclipse-vertx/vert.x/blob/6a8aefb6ec84278774d2c03fdae83fdf1deafded/src/main/java/io/vertx/core/impl/VertxImpl.java#L104: it means that if I won't expose something public to check if the immutable sys prop is there, with an exception, this one will crash the application while starting

Reply:
Any suggestion @normanmaurer ? do you want me to add an exception for this?

Reply:
I am not sure ... @chrisvest @trustin thoughts ?

Reply:
Thinking out loud, what about:
1. Throwing an unsupported exception if the new level is different than the immutable one. 
2. Always require `io.netty.leakDetection.level` to be set when setting `io.netty.immutableResourceLeakDetection`.